ooniprobe/Utility/UIDevice-Hardware.m

```diff
@@ -0,0 +1,31 @@
+#include <sys/sysctl.h>
+#import "UIDevice-Hardware.h"
+
+@interface UIDevice (Hardward)
+
+- (NSString *)modelNameForModelIdentifier:(NSString *)modelIdentifier;
+
+@end
+
+@implementation UIDevice (Hardware)
+
+- (NSString *)getSysInfoByName:(char *)typeSpecifier
+{
+    size_t size;
+    sysctlbyname(typeSpecifier, NULL, &size, NULL, 0);
+
+    char *answer = malloc(size);
```

Comment:
@bassosimone probably knows this better, but I believe you need to check if the return value of malloc is null to verify if you have allocated memory successfully.

Reply:
I know nothing about this code. It's the same code of the library, of iOS cookbook and of many github answers so I guess it's the way to go 🤷‍♂️

Reply:
So, it really depends on the VMM. Generic code _should_ check the return value of `malloc` and call `abort` if it's `NULL`. On Linux, and possibly also on macOS, the VMM does not fail in its default configuration and later, if there's not enough core, you get a segfault. My suggestion is to check the return value and `abort` if it's `NULL`.

Reply:
https://books.google.ru/books?id=0DdbblZ6NlYC&pg=PT37&lpg=PT37&dq=%22hw.machine%22+cookbook&source=bl&ots=oTYPC7igAQ&sig=ACfU3U3nZWK4v4ZcvbSkKuo-BstTwPA9FQ&hl=it&sa=X&ved=2ahUKEwjHu-Os2I7hAhWv-ioKHf7oAe8Q6AEwAHoECAEQAQ#v=onepage&q=%22hw.machine%22%20cookbook&f=false

Reply:
Not checking the return value of `malloc` is not a good programming practice. It leads to headaches when porting code to other systems. High performance code may decide not to do that if the platform where this happens is clearly one where `malloc` does not return `NULL` in case of failure. This is not the case. Please, check the return value of `malloc`.